runtime/tests/Test_NullQubit.cpp

```diff
@@ -414,6 +452,99 @@ TEST_CASE("test AllocateQubits generates a proper std::vector<QubitIdType>", "[N
     }
 }
 
+TEST_CASE("Test allocation of multiple registers", "[NullQubit]")
+{
+    std::unique_ptr<NullQubit> sim = std::make_unique<NullQubit>();
+
+    auto &&reg1 = sim->AllocateQubits(1);
+    auto &&reg2 = sim->AllocateQubits(2);
+
+    CHECK(reg1.size() == 1);
+    CHECK(reg2.size() == 2);
+    CHECK(sim->GetNumQubits() == 3);
+}
+
+TEST_CASE_METHOD(NullQubitRuntimeFixture, "Test insertion of qubit into register", "[NullQubit]")
+{
+    // Allocate register with three qubits, [0, 1, 2]
+    QirArray *reg = __catalyst__rt__qubit_allocate_array(3);
+
+    auto reg_vec_before = *reinterpret_cast<std::vector<QubitIdType> *>(reg);
+
+    CHECK(reg_vec_before[0] == 0);
+    CHECK(reg_vec_before[1] == 1);
+    CHECK(reg_vec_before[2] == 2);
+
+    // Release qubit 1; note that array `reg` is still [0, 1, 2]
+    __catalyst__rt__qubit_release(reinterpret_cast<QUBIT *>(1));
```

Comment:
```suggestion
    // Release qubit 1; note that array `reg` is still [0, 1, 2]
    __catalyst__rt__qubit_release(reinterpret_cast<QUBIT *>(reg_vec_before[1]));
```

let's get the ID from the register array properly instead of hard coding the ID : ) I know the ID generation scheme for null qubit will probably never change, but I always like to reduce tech debt even in small places whenever possible. Especially in tests, since random unrelated tests failing when I'm developing something else always gives me anxiety :laughing: